src/mapgen/mapgen.cpp

```diff
@@ -551,18 +550,24 @@ void Mapgen::spreadLight(v3s16 nmin, v3s16 nmax)
 
 				u8 light = n.param1;
 				if (light) {
-					lightSpread(a, v3s16(x,     y,     z + 1), light);
-					lightSpread(a, v3s16(x,     y + 1, z    ), light);
-					lightSpread(a, v3s16(x + 1, y,     z    ), light);
-					lightSpread(a, v3s16(x,     y,     z - 1), light);
-					lightSpread(a, v3s16(x,     y - 1, z    ), light);
-					lightSpread(a, v3s16(x - 1, y,     z    ), light);
+					const v3s16 p(x, y, z);
+					// spread to all 6 neighbor nodes
+					for (const auto &dir : g_6dirs)
+						lightSpread(a, queue, p + dir, light);
```

Comment:
One might ask, of course, why here `p, light` isn't added to the queue instead. But that would just mean that those values are added to the queue and removed again a bit later in the same order.